src/explore-education-statistics-admin/src/utils/ckeditor/customUploadAdapterPlugin.ts

```diff
@@ -9,8 +9,15 @@ export default function customUploadAdapterPlugin(
   onUpload: ImageUploadHandler,
   onCancel?: ImageUploadCancelHandler,
 ) {
-  return class CustomUploadAdapterPlugin {
-    constructor(editor: Editor) {
+  return class CustomUploadAdapterPlugin implements Plugin {
+    static get pluginName(): PluginName {
+      return 'CustomUploadAdapter';
+    }
+
+    constructor(
+      editor: Editor,
+      readonly pluginName: PluginName,
```

Comment:
Ah, apologies, looks we need to change the types a bit more:

```ts
// types/ckeditor.ts

export interface PluginClass {
  get pluginName(): PluginName;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Plugin {}
```

Then we can use `PluginClass` for the adapter so we don't need to provide this as a property e.g.

```ts
return class CustomUploadAdapterPlugin implements PluginClass {
  get pluginName(): PluginName {
    return 'CustomUploadAdapter';
  }

  constructor(editor: Editor) {
    const fileRepository =
      editor.plugins.get<FileRepository>('FileRepository');

    fileRepository.createUploadAdapter = loader => {
      return new CustomUploadAdapter(loader, onUpload, onCancel);
    };
  }
};
```

We also need to change `CustomEditor` to use this as well:

```ts
// types/ckeditor-ambient.d.ts

interface CustomEditor extends EditorClass {
    builtinPlugins?: PluginClass[];
}
```

Reply:
Cool, sorted now,